packages/signature-controller/src/SignatureController.ts

```diff
@@ -880,4 +897,67 @@ export class SignatureController extends BaseController<
 
     return networkClient.configuration.chainId;
   }
+
+  #decodePermitSignatureRequest(
+    signatureRequestId: string,
+    request: OriginalRequest,
+    chainId: string,
+  ) {
+    // Code below will invoke signature request decoding api for permits
+    if (!this.#decodingApi) {
+      return;
+    }
+    let decodedRequest: DecodedRequestInfo;
+    try {
+      const { primaryType } = JSON.parse(request.params?.[1] ?? '');
+      if (primaryType === 'Permit') {
+        this.#updateMetadata(signatureRequestId, (draftMetadata) => {
+          draftMetadata.decodedRequest = 'IN_PROGRESS';
```

Comment:
Rather than having a property with an alternating type, should we provide an explicit `decodingLoading` property instead and leave this undefined?

Reply:
Updated.